### PR TITLE
RegexpQueryParser takes a String as value like its Builder

### PR DESCRIPTION
**Pull request by alexksikes**
Relates #11896

### CODE REVIEW

**Comment by javanna** [comment]
don't we want to change the mappers as well to accept a String instead of an Object?

**Comment by alexksikes** [comment]
@javanna Thanks for the review. I changed the mapper as well.

File: core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         currentFieldName = parser.currentName();
                     } else {
                         if ("value".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.textOrNull();
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java]: not sure whether we want to use text() or textOrNull() here

**Comment by javanna** [comment]
looks good, @jpountz can you please have a look too? I had a question above ;)

File: core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java (hunk 1 above)
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java]: Alex's comment on the prefix query that we need to check for null anyway and can have a better error message this way makes sense to me

**Comment by jpountz** [comment]
LGTM

**Pull request closed without merging**